inc/functions-cookies.php

```diff
@@ -0,0 +1,75 @@
+<?php
+// These functions are going to be moved into the plugin
+// upon cross site roll-out
+
+function delete_GA_cookies() {
+	$domain = 'nationalarchives.gov.uk';
+	$cookie_list = ['_ga', '_gid', '_gat_UA-2827241-22', '_gat_UA-2827241-1'];
+    
+	if (isset($_SERVER['HTTP_COOKIE'])) {
+		$cookies = explode(';', $_SERVER['HTTP_COOKIE']);
+		foreach($cookies as $cookie) {
+			$parts = explode('=', $cookie);
+			$name = trim($parts[0]);
+			foreach($cookie_list as $single_cookie) {
+				if($name == $single_cookie) {
+                    unset( $_COOKIE[$name] );
+                    setcookie($name, '', time()-10000000, '/', '.' . $domain);
+                    setcookie($name, '', time()-10000000, '/', $domain);
+                    setcookie($name, '', time()-10000000, '/');
+                }
+			}
+		}
+	}
+}
+
+function handle_GA_script(string $global_cookie) {
+    $siteUrl = site_url();
+
+    if (strpos($siteUrl, 'latin') !== false) {            
+        if(isset($_COOKIE[$global_cookie])) {
+            $cookie = $_COOKIE[$global_cookie];
+            $clean_cookie = preg_replace('/\\\\/', '', $cookie);
+            $cookies_policy_to_obj = json_decode( $clean_cookie );
+            if($cookies_policy_to_obj->usage == true) { 
+                include 'gtm-script.php';
+            } 
+        }
+
+    } else { 
+        include 'gtm-script.php';
+    }
+}
+
+// New Cookie Consent
+//! This part of the code is going to be moved into the plugin
+if ( isset($_POST['measure-website-use']) ) {
+	if (function_exists('delete_GA_cookies')) {
+		add_action( 'init', 'delete_GA_cookies' );
+   }
+}
+
+// Remove cookies if available when coming from Latin
+function remove_cookies_on_page_load() {
+	$siteUrl = site_url();
+	$global_cookie = 'cookies_policy';
+	
+	if (strpos($siteUrl, 'latin') !== false) {            
+		if(isset($_COOKIE[$global_cookie])) {
+			$cookie = $_COOKIE[$global_cookie];
+			$clean_cookie = preg_replace('/\\\\/', '', $cookie);
+			$cookies_policy_to_obj = json_decode( $clean_cookie );
+			if($cookies_policy_to_obj->usage === false) { 
```

Comment:
Let's add a quick check here for this property (just in case the JSON structure should change at some point in the future and we encounter an undefined property warning). Something like `property_exists()` https://www.php.net/manual/en/function.property-exists.php

Reply:
I tried `if(property_exists((object) $cookies_policy_to_obj, 'usage') {` but it doesn't like it. I feel like it's waiting for the object to be initialised from a `Class`....

Reply:
Ok, I worked on the second try